### DISCUSSION TITLE
Moving mesh on NEKRS

### DISCUSSION THREAD

**Discussion by chanleonzen** [Nek5000/nekRS#418] [Q&A]
Hello there,

I was wondering if there is a moving mesh example for NEKRS similar to mv_wall in NEK5000? 
Thank you.

Kind regards,
Leon

**Comment by yslan** [comment]
There is
https://github.com/Nek5000/nekRS/tree/master/examples/mv_cyl
similar to
https://github.com/Nek5000/NekExamples/blob/master/mv_cyl/mv_cyl.usr

**Comment by chanleonzen** [comment]
Thanks Yu-Hsiang. That’s great. 
I must have missed this when I downloaded nekrs a while back.

Kind regards,
Leon

**Reply by chanleonzen** [reply]
I have run the case and it works but I do not see the mesh deformation as the coordinates are only dumped in the 1st field file. I need to change 
  platform->options.setArgs("CHECKPOINT OUTPUT MESH", "FALSE");
in the source code. However, I was wondering if there is a switch in the .par file which I can use? 
Thank you.

**Comment by stgeke** [comment]
There is a par key `writeToFieldFile`. Run `nekrs --help par` for more details.

**Reply by chanleonzen** [reply]
Thank you for the tips and advice.

**Reply by chanleonzen** [reply]
I am getting this error
detected par file errors:
        unknown key: mesh::writetofieldfile
        
I ran nekrs --help par but did not find the "writeToFieldFile" key either. This is for v21.1. Have I done something silly?

**Reply by stgeke** [reply]
Sorry it's something we added recently. A preview of the next upcoming release is available in the `next` branch.

**Reply by chanleonzen** [reply]
All good. Looking forward to the upcoming release and appreciate your help and support.

**Comment by chanleonzen** [comment]
I've been trying to modify the mv_cyl example to suit my case but I am having a bit of difficulty. Firstly, the example uses the low Mach solver and I am trying to change it to incompressible flow. So I've removed 

`udf.properties = &uservp;`
 `udf.sEqnSource = &userq;`

 `lowMach::setup(nrs, P_GAMMA);`
 `udf.div = &qtl;` 

from UDF_Setup and 

`userVpKernel
lowMach`

from UDF_LoadKernels but I am getting issues with solving for velocity and the simulations explodes. Eventually, I would like to setup a channel flow with moving walls. 

Thank you
